papa/src/main/java/papa/InteractionStartInput.kt

```diff
@@ -0,0 +1,8 @@
+package papa
+
+import kotlin.time.Duration
+
+class InteractionStartInput(
+  val inputData: Any,
```

Comment:
Do we want papa responsible for `inputData` accounting?  In the integration on register we'd probably carry any data through `InteractionEvent`

Reply:
So yes, the general idea is that we have the custom timestamp passed in as a property of the sent event, and then in the interaction rule client, we read that and pass it in to startInteraction. In that specific case, we don't need `inputData` at all (so now that I think of it is should probs be `inputData: Any? = null`).

The `inputData` field is here for the receiving end, where we get the `InteractionResult` and pull out the input details

E.g. look for `interactionInput` usages here (Square internal link):  https://github.com/squareup/android-register/blob/master/common/ui-latency/impl/src/main/java/com/squareup/uilatency/InteractionLatencyAnalytics.kt


```
ui_latency_trigger_frames = result.interactionInput?.framesSinceDelivery?.toLong()?: -1,
```

=> this needs access to the DeliveredInput which in this PR we shove inside `InteractionStartInput.inputData`


```
  private val InteractionResultData<InteractionEvent>.interactionType: String
    get() = interactionInput?.let {
      when (val inputEvent = it.event) {
        is MotionEvent -> "tap"
        is KeyEvent -> {
          "key ${inputEvent.name}"
        }

        else -> error("Unexpected input event type $inputEvent")
      }
    } ?: "unknown"
```

=> here we're mostly looking for a string description of the event.

```
  private val InteractionResultData<InteractionEvent>.triggerDuration: Duration?
    get() {
      val firstEvent = sentEvents.first()
      val triggerUptime = interactionInput?.eventUptime
      return triggerUptime?.let { firstEvent.uptime - it }
    }
```

=> This one only uses `eventUptime` which is carried over as `inputUptime`, so that one doesn't need `inputData`.

I'm thinking maybe I should turn InteractionStartInput into an interface implemented by DeliveredInput, that can be implemented by our custom things, and then I can remove `inputData` from it and just have `inputUptime`

Reply:
done!